File: CanSum/canSum.cpp
#include<bits/stdc++.h>
using namespace std;

bool CanSum(int targetSum, const vector<int>& numbers) {
    if (targetSum == 0) return true;
    if (targetSum < 0) return false;

    for (int num : numbers) {
        int remainder = targetSum - num;
        if (CanSum(remainder, numbers)) {
            return true;
        }
    }
    return false;
}

int main() {
    cout << boolalpha;
    cout << CanSum(7, {5, 3, 4, 7}) << endl;
    cout << CanSum(7, {2}) << endl;
    cout << CanSum(300, {7, 14}) << endl;
    return 0;
}

File: CanSum/canSum.js
const CanSum = (targetSum, numbers) =>
{
    if (targetSum ===0) return true
    if (targetSum <= 0) return false

    for (let num of numbers)
    {
        const remainder = targetSum - num
        if (CanSum(remainder, numbers) === true)
        {
            return true
        }
    }
    return false
}

console.log(CanSum(7,[5,3,4,7]))
console.log(CanSum(7,[2]))
console.log(CanSum(300,[7,14]))